src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -75,6 +75,28 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
+        if (this._redirectFlowIsTrue()) {
+            let nonce: { id: string };
+            try {
+                nonce = JSON.parse(this._paymentMethod.clientToken);
+            } catch (error) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            await this._prepareForReferredRegistration(payment.methodId, nonce.id);
+
+            try {
+                return await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId: payment.methodId, paymentData: { nonce: nonce.id }}));
+            } catch (error) {
+                if (error instanceof RequestError && error.body.status === 'additional_action_required') {
+                    return new Promise(() => {
```

Comment:
this method is `async` so you don't need to return a promise. You can simply do
```js
const { redirect_url } = error.body.additional_action_required.data;
window.location.replace(redirect_url);

return;
```
 
Also, object destructuring is recommended to improve readability.

Reply:
I kept the returned promised due to types, If I don't return the promise it will create a mismatch error with the expected `Promise<InternalCheckoutSelectors>` of the execute function.

Reply:
That's fine, you can keep the return type `Promise<InternalCheckoutSelectors>`. When the method is `async`, it will automatically wrap any returned value in a Promise, so it's redundant to manually wrap the returned value in a Promise.

Reply:
Following your recommendation i would have to delete `return` because it returns void which causes a mismatch with the return type of execute, so if I delete it and keep only the window.location.replace it will throw this error before doing the actual redirect. 

<img width="1388" alt="Screen Shot 2021-04-26 at 20 34 37" src="https://user-images.githubusercontent.com/4503787/116172110-bd577a80-a6cf-11eb-9812-67d4a04d8205.png">

Any suggestion related to that?

Reply:
Right, I think I understand. What you are actually returning here is `Promise<never>` which is a promise that never resolves.

one thing you could do but doesn't really matter much is
```js
const { redirect_url } = error.body.additional_action_required.data;
window.location.replace(redirect_url);

return new Promise();
```

There's really no need to wrap that code around a promise.

Reply:
I will keep it right now as it is, If a return the empty promise I still can get errors
 ```ERROR in /Users/estefaniaocampo/dev/cloud-dev-vm/bcappvm/codebases/microapps/checkout-sdk-js/src/payment/strategies/zip/zip-payment-strategy.ts
[tsl] ERROR in /Users/estefaniaocampo/dev/cloud-dev-vm/bcappvm/codebases/microapps/checkout-sdk-js/src/payment/strategies/zip/zip-payment-strategy.ts(99,28)
      TS2554: Expected 1 arguments, but got 0.```

Thenks for the review.